File: c++/stackreversestring.cpp
/*
 * Frank Petrilli | frank@petril.li | frank.petril.li
 * Language: C++
 * String reverser using a stack.
 */

#include <stdio.h>
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main(int argc, char** argv)
{
	stack<char> s;

	string input;

	if (argc == 2)
	{
		input = argv[1];
	}
	else
	{
		cout << "Enter a word: ";
		getline(cin,input);
	}

	for (int i = 0; i < input.size(); i++) {
		s.push(input.at(i));
	}

	while (!s.empty()) {
		cout << s.top();
		s.pop();
	}
	cout << "\n";
	return 0;
}

File: c#/StackReverseString.cs
/*
 * Frank Petrilli | frank@petril.li | frank.petril.li
 * Language: C#
 * Creates a Stack, feeds a line of text into it char by char, then spits it back out on the console.
 */

using System;
using System.Collections;

public class StackReverseString
{
	public static void Main(string[] args)
	{
		Stack s = new Stack();
		string input;
		if (args.GetLength(0) < 1) {
		       	Console.Write("Enter a string: ");
			input = Console.ReadLine();
		} else {
			input = args[0];
		}
		for (int i = 0; i < input.Length; i++)
		{
			s.Push(input[i]);
		}

		while (s.Count != 0)
		{
			Console.Write(s.Pop());
		}

		Console.WriteLine();

		if (args.GetLength(0) < 1) {
			Console.ReadLine();
		}
	}
}

